protocol-units/execution/opt-executor/src/executor/mod.rs

```diff
@@ -40,20 +40,17 @@ impl Executor {
 		Arc::clone(&self.db().reader)
 	}
 
-	pub fn decrement_transactions_in_flight(&self, count: u64) {
-		// fetch sub mind the underflow
-		// a semaphore might be better here as this will rerun until the value does not change during the operation
-		self.transactions_in_flight
-			.fetch_update(Ordering::Relaxed, Ordering::Relaxed, |current| {
-				info!(
-					target: "movement_timing",
-					count,
-					current,
-					"decrementing_transactions_in_flight",
-				);
-				Some(current.saturating_sub(count))
-			})
-			.unwrap_or_else(|_| 0);
+	pub async fn decrement_transactions_in_flight(&self, count: u64) {
```

Comment:
This method, and the `decrement_transaction_in_flight` method upwards of it, does not need to be async any more.